design/user_types.go

```diff
@@ -38,6 +38,71 @@ var UpdateWorkItemPayload = a.Type("UpdateWorkItemPayload", func() {
 	a.Required("type", "fields", "version")
 })
 
+// UpdateWorkItemJSONAPIPayload defines top level structure from jsonapi specs
+// visit : http://jsonapi.org/format/#document-top-level
+var updateWorkItemJSONAPIPayload = a.Type("UpdateWorkItemJSONAPIPayload", func() {
+	a.Attribute("data", workItemDataForUpdate)
+	a.Required("data")
+})
+
+// WorkItemDataForUpdate defines how an update payload will look like
+var workItemDataForUpdate = a.Type("WorkItemDataForUpdate", func() {
+	a.Attribute("type", d.String, func() {
+		a.Enum("workitems")
+	})
+	a.Attribute("id", d.String, "ID of the work item which is being updated", func() {
+		a.Example("42")
+	})
+	a.Attribute("attributes", a.HashOf(d.String, d.Any), func() {
+		a.Example(map[string]interface{}{"version": "1", "system.state": "new", "system.title": "Example story"})
+	})
+	a.Attribute("relationships", workItemRelationships)
+	// relationships must be required becasue we MUST have workItemType during PATCh
+	a.Required("type", "id", "attributes")
+})
+
+// WorkItemRelationships defines only `assignee` as of now. To be updated
+var workItemRelationships = a.Type("WorkItemRelationships", func() {
+	a.Attribute("assignee", RelationAssignee, "This deinfes assignees of the WI")
+	a.Attribute("baseType", RelationBaseType, "This defines type of Work Item")
+	// baseType relationship must present while updating work item
+})
+
+// RelationAssignee is a top level structure for assignee relationship
+var RelationAssignee = a.Type("RelationAssignee", func() {
+	a.Attribute("data", AssigneeData)
+	a.Required("data")
+})
+
+// AssigneeData defines what is needed inside Assignee Relationship
+var AssigneeData = a.Type("AssigneeData", func() {
+	a.Attribute("type", d.String, func() {
+		a.Enum("identities")
+	})
+	a.Attribute("id", d.String, "UUID of the identity", func() {
+		a.Example("6c5610be-30b2-4880-9fec-81e4f8e4fd76")
+	})
+	a.Required("type")
```

Comment:
As mentioned [here](https://github.com/almighty/almighty-ui/issues/348#issue-191438557), I am using `nil` valud of ID as `remove assignee`. Currently I am not following [this](http://jsonapi.org/format/#crud-updating-relationships)  to update relationships exclusively. These will come later, I have already started work on the same. tracked #513